tests/integration.rs

```diff
@@ -0,0 +1,54 @@
+use svd_parser as svd;
+/// Testing SVDs by (de)-serializing them, ideally all of those on:
+/// https://github.com/posborne/cmsis-svd
+
+#[test]
+fn read_and_write() {
+    // Load an SVD XML to a string in memory
+    let svd_str = include_str!(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/tests/data/esp8266.svd"
+    ));
+
+    // Parse SVD device
+    let device = svd::parse(svd_str).expect("Failed to parse the SVD file into Rust structs");
+
+    // Write (encode/serialize) SVD
+    let device_svd = svd::encode(&device).unwrap();
+
+    // Read and Write (deserialize) SVD again
+    let device_svd_deserialized = svd::encode(&svd::parse(&device_svd).unwrap()).unwrap();
+
+    assert_eq!(device_svd, device_svd_deserialized)
+}
+
+// fn diff_svds(orig: Path, dest: Path) -> Result<Diff> {
```

Comment:
@MarcoIeni, Are you exploring https://github.com/MarcoIeni/delta for SVD diffing currently?

Reply:
no, I did some work on delta just for fun, not svd related reasons :P